### DISCUSSION TITLE
How to retrigger generateMetadata without page refresh?

### DISCUSSION THREAD

**Discussion by nmelentjevs** [vercel/next.js#55507] [Help] [answered]
### Summary

I have an app that requires page titles to be updated according to token price changes, I could generate the initial title using generateMetadata method, however, I didn't find any documentation on how to regenerate metadata dynamically with an interval for example, or manually retrigger it.

Any help is appreciated, thank you!

### Additional information
Using App Router
```
"next": "13.4.12"
"react": "18.2.0"
```

**Comment by icyJoseph** [comment]
I did something like this at page level:

```tsx
import type { Metadata } from "next";

export const revalidate = 10;

export const generateMetadata = (): Metadata => ({
  title: `${Date.now()}`,
  description: "Generated by create next app",
});

export default function Home() {
  return (
    <main className="flex min-h-screen flex-col items-center justify-between p-24">
      {Date.now()}
    </main>
  );
}
```

And I see it changing.

**Reply by nmelentjevs** [reply]
Had copied your exact code just to make sure: still doesn't update for me. I am on the 13.4.12 version.

**Reply by icyJoseph** [reply]
Maybe there's something else going on?

https://github.com/vercel/next.js/assets/21013447/54b7c5ee-fa20-4b89-b7c1-ddb390438d6d

**Reply by nmelentjevs** [reply]
With page refresh the title always regenerates, but I am looking to keep retriggering the title update without page refreshing

**Reply by icyJoseph** [reply, marked as answer]
Aha, mmm like after a user action and such? 

I mean title, can be changed as result of user actions and to a certain degree, use-effects, by just setting the `document.title` property to something else.

```tsx
<button onClick={() => {document.title = 'foo-bar'}}>add to cart</button>
```

**Reply by icyJoseph** [reply]
I guess you want something like this?

https://github.com/vercel/next.js/assets/21013447/797c70fa-afe0-482c-9dc6-b994f901cd5e

```tsx
"use client";

import { useRouter } from "next/navigation";

export const Refresh = () => {
  const router = useRouter();
  return <button onClick={() => router.refresh()}>Refresh</button>;
};
```

And use `Refresh` on your page component.

**Reply by nmelentjevs** [reply]
That... actually solves it, appreciate it! I spent way too much time inside next docs looking for a way to use in-built methods for this, this solution is much easier - thank you!

**Reply by icyJoseph** [reply]
Awesome! Please don't forget to mark as answered :) Glad to have helped!

**Reply by GabenGar** [reply]
Did the `<Head/>` component of the `pages` router work the same way as setting the `document.title` value? The docs for [`document.title`](https://developer.mozilla.org/en-US/docs/Web/API/Document/title) don't say anything about sanitizing values the same way a section in [`Node.textContent`](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent#differences_from_innerhtml) does. I assume the `<Head/>` component used the react's built-in sanitizer at least.
